### ISSUE TITLE
More complex benchmarks

### ISSUE THREAD

**Issue by martinfleis** [kadyb/vector-benchmark#13]
Hey @kadyb,

would you be up for expanding the benchmarks with more complex tasks? I have a felling that those you have in here now, especially for buffer and distance, can be a bit misleading. Since both of them are based on points only, some libraries (terra, sf) have a special fast-path for them. I would be interested in more complex options, like distance between polygons and buffering of polygons. Right now, these tests sort of only check if the library has a fast-path implemented for these or not.

**Comment by kadyb** [comment]
Sure! I will try to add the tasks you suggested. If you have more ideas for tasks, let me know.

**Comment by martinfleis** [comment]
This is an example of what I was talking about regarding the fast-path distance algo - https://github.com/shapely/shapely/issues/2038

**Comment by kadyb** [comment]
Maybe we should report this in the GEOS repository? `sf` uses a base function from R, i.e. `dist()`, but this is not the fastest way. Even better results are provided by the `Rfast::Dist()`, which is written in C++.

**Comment by martinfleis** [comment]
> Maybe we should report this in the GEOS repository? 

Yeah, @theroggy suggested the same in the shapely issue. Will do.

> sf uses a base function from R, i.e. dist(), but this is not the fastest way

I don't think so, see https://github.com/r-spatial/sf/blob/32aa6c8c53005245493540b1359bc35fe50f0115/R/geom-measures.R#L200-L203 But I know little about the performance of R world.

**Comment by kadyb** [comment]
`by_element` is FALSE, so this path: https://github.com/r-spatial/sf/blob/32aa6c8c53005245493540b1359bc35fe50f0115/R/geom-measures.R#L207-L208

**Comment by martinfleis** [comment]
Ah, okay. You see I know nothing about R :D.

The GEOS issue - https://github.com/libgeos/geos/issues/1066

**Comment by kadyb** [comment]
After the workshop in Prague, you will be specialist in R and Julia 😄

**Comment by theroggy** [comment]
They won't be directly usable, but you can get some inspiration for benchmarks for some more complex spatial operations here: https://github.com/geofileops/geobenchmark/blob/main/results_vector_ops/GeoBenchmark.png
